### PR TITLE
Store tags to a persistent SQL store

### PR DESCRIPTION
**Pull request by pudo**
This is a draft, I would appreciate your feedback @sunu. It stores the tags to SQL, but does not store which crawler a tag belongs to explicitly, meaning there can be no tag flushing for a whole dataset. 

I'm trying to decide how big a fault/downside that is. What we gain is simplicity. 

Update: so I've added support for `tags.delete(prefix=crawler.name)` to `servicelayer`.

### CODE REVIEW

File: memorious/logic/context.py [hunk 1]
```diff
@@ -146,7 +145,7 @@ def skip_incremental(self, *criteria):
         if self.check_tag(key):
             return True
 
-        self.set_tag(key, None)
+        self.set_tag(key, 'inc')
```
Comment (sunu) [inline memorious/logic/context.py:148]: What's `inc`?

**Comment by sunu** [comment]
Looks great! I'm keen to move `Events` to the sql store too. Should that be done through servicelayer too?

**Comment by pudo** [comment]
I'm keen to understand how that would interact with the `servicelayer.reporting` module written by @simonwoerpel - it produces the same sort of data but has no persistence (yet).

**Comment by sunu** [comment]
> I'm keen to understand how that would interact with the `servicelayer.reporting` module written by @simonwoerpel - it produces the same sort of data but has no persistence (yet).

I'll take a look at that to get an idea how well it fits with the data from memorious events.

Besides that, is this ok to merge or do we need to make any servicelayer related config changes in the prod configs before this is released?

**Comment by pudo** [comment]
This should be ready for prod, I've tested the implementation mainly with Aleph but it seems stable. 

What I'd propose we do in terms of upgrade is to a) wait till there's few crawlers running, b) Remove all the crawlers deployments, c) delete crawlers-redis from disk and remove the backup function for it - not needed any more, d) bring this up fresh.

**Pull request merged**